squirrel_safety/src/squirrel_safety/safety.py

```diff
@@ -0,0 +1,51 @@
+import rospy
+
+from squirrel_safety_msgs.msg import Safety
+from std_msgs.msg import Bool
+
+class SquirrelSafety(object):
+    def __init__(self, airskin_topic, bumper_topic, wrist_topic):
+        while rospy.get_time() == 0.0:
+            pass
+
+        rospy.loginfo(rospy.get_caller_id() + ': starting up')
+
+        self.wrist_sub = rospy.Subscriber(wrist_topic, Bool, self.wrist_callback)
+        self.bumper_sub = rospy.Subscriber(bumper_topic, Bool, self.bumper_callback)
+        self.airskin_sub = rospy.Subscriber(airskin_topic, Bool, self.airskin_callback)
+        self.safety_pub = rospy.Publisher('/squirrel_safety', Safety, queue_size=10)
+
+        rate = rospy.Rate(10)
+        self.safety_msg = Safety()
+        self.safety_msg.emergency_state = False
+        self.safety_msg.airskin_stop = False
+        self.safety_msg.wrist_stop = False
+        self.safety_msg.bumper_stop = False
+
+        while not rospy.is_shutdown():
+            if self.safety_msg.wrist_stop or self.safety_msg.bumper_stop or self.safety_msg.airskin_stop:
+                self.safety_msg.emergency_state = True
+            self.safety_pub.publish(self.safety_msg)
+            self.safety_msg.emergency_state = False
+            rate.sleep()
+
+
+    def airskin_callback(self, msg):
+        if msg:
```

Comment:
Oh snap, got me on that one! You're right, as soon as the object is not None, it's true, so in this check it's always true. Regarding your other question, that's a good point. One way would be to check whether the topic is there or the rate on the topic. But how would you handle that? 

Regarding your work-around: yep, sounds totally reasonable. Will you do it or shall I do it?